internal/controllers/gateway/gateway_utils_test.go

```diff
@@ -130,15 +130,25 @@ func TestGetListenerStatus_no_duplicated_Detached_condition(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, statuses, 1, "only one listener status expected as only one listener was defined")
 	listenerStatus := statuses[0]
-	assertOnlyOneConditionOfType(t, listenerStatus.Conditions, gatewayv1beta1.ListenerConditionAccepted)
+	assertOnlyOneConditionOfTypes(t, listenerStatus.Conditions,
+		gatewayv1beta1.ListenerConditionAccepted,
+		gatewayv1beta1.ListenerConditionReady,
+		gatewayv1beta1.ListenerConditionProgrammed,
+		gatewayv1beta1.ListenerConditionResolvedRefs,
+		gatewayv1beta1.ListenerConditionConflicted,
+	)
 }
 
-func assertOnlyOneConditionOfType(t *testing.T, conditions []metav1.Condition, typ gatewayv1beta1.ListenerConditionType) {
-	conditionNum := 0
-	for _, condition := range conditions {
-		if condition.Type == string(typ) {
-			conditionNum++
+func assertOnlyOneConditionOfTypes(t *testing.T, conditions []metav1.Condition, types ...gatewayv1beta1.ListenerConditionType) {
+	conditionsNum := make([]int, len(types))
+	for i, t := range types {
+		for _, condition := range conditions {
+			if condition.Type == string(t) {
+				conditionsNum[i]++
+			}
 		}
 	}
-	assert.Equal(t, 1, conditionNum)
+	for _, n := range conditionsNum {
+		assert.LessOrEqual(t, n, 1)
```

Comment:
This might be made a bit more descriptive. In case this fails it will be hard to track down which condition was duplicated.